src/legend.ts

```diff
@@ -0,0 +1,237 @@
+        else {
+            defs.appendChild(marker)
+            marker.appendChild(polygon)
+        }
+       
+        arrowSVG.appendChild(line)
+
+        return arrowSVG
+    }
+
+    function getTableElement(arrowName: string) {
+        let arrow = genSVGArrow("#2196f3", false, false, "n/a", "n/a", "url(#arrowhead1)", "arrowhead1", "n/a") // default is stateless arrow 
+
+        if (arrowName == "dataStateful") {
+            arrow = genSVGArrow("#2196f3", true, false, "n/a", "n/a", "url(#arrowhead)", "arrowhead", "n/a")
+        }
+
+        else if (arrowName == "state") {
+            arrow = genSVGArrow("red", false, false, "n/a", "n/a", "url(#endarrow-state)", "endarrow-state", "n/a")
+        }
+
+        else if (arrowName == "network") {
+            arrow = genSVGArrow("#71c94f", false, true, "url(#startarrow)", "url(#endarrow)", "n/a", "startarrow", "endarrow")
+        }
+
+        return arrow 
+    }
+
+    function getTableText(elemName: string) {
+        let td = document.createElement("td")
+
+        if (elemName == "dataStateless") {
+            td.innerText="Data (stateless)"
+        }
+
+        else if (elemName == "dataStateful") {
+            td.innerText="Data (stateful)"
+        }
+
+        else if (elemName == "state") {
+            td.innerText="#State"
+        }
+
+        else if (elemName == "network") {
+            td.innerText="Network"
+        }
+
+        else if(elemName == "clock") {
+            td.innerText="Time"
+        }
+
+        else if (elemName == "module") {
+            td.innerText="Module"
+        }
+
+        else {
+            td.innerText="Sensor"
+        }
+
+        return td
+    }
+
+    function appendArrow(arrowName: string) {
+        let td = document.createElement("td")
+
+        let element = getTableElement(arrowName)
+
+        td.appendChild(element)
+
+        return td
+    }
+
+    function appendTableElement(elemName: string) {
+        let td = document.createElement("td")
+        let color = "transparent"
+
+        if (elemName == "sensor") {
+            color = "#777"
+        }
+
+        let svg = document.createElementNS("http://www.w3.org/2000/svg", "svg")
+        svg.setAttribute("viewBox", "0 0 220 100")
+        svg.setAttribute("xmlns", "http://www.w3.org/2000/svg")
+
+        let rect = document.createElementNS("http://www.w3.org/2000/svg", "rect")
+        rect.setAttribute("x", "35")
+        rect.setAttribute("y", "5")
+        rect.setAttribute("width", "150")
+        rect.setAttribute("height", "90")
+        rect.setAttribute("stroke", "black")
+        rect.setAttribute("fill", color)
+        rect.setAttribute("stroke-width", "3")
+
+        svg.appendChild(rect)
+        td.appendChild(svg)
+
+        return td
+    }
+
+    function appendIMG() {
+        let td = document.createElement("td")
+
+        let img = document.createElement("img")
+        img.setAttribute("src", "file:///Users/mjaylene/karl-ui/src/img/clock.png")
```

Comment:
This path doesn't work on my computer because I don't have a folder called `Users/mjaylene/...`. :)

Reply:
I defined a different path to the image that does not use my own personal folder.